editors/code/src/highlighting.ts

```diff
@@ -32,36 +65,44 @@ export class Highlighter {
             tag: string,
             textDecoration?: string
         ): [string, vscode.TextEditorDecorationType] => {
-            const color = new vscode.ThemeColor('ralsp.' + tag);
-            const decor = vscode.window.createTextEditorDecorationType({
-                color,
-                textDecoration
-            });
-            return [tag, decor];
+            const scope = scopes.find(tag)
+
+            if (scope) {
+                const decor = createDecorationFromTextmate(scope);
+                return [tag, decor];
+            }
+            else {
+                const color = new vscode.ThemeColor('ralsp.' + tag);
+                const decor = vscode.window.createTextEditorDecorationType({
+                    color,
+                    textDecoration
+                });
+                return [tag, decor];
+            }
         };
 
         const decorations: Iterable<
             [string, vscode.TextEditorDecorationType]
         > = [
-            decoration('comment'),
-            decoration('string'),
-            decoration('keyword'),
-            decoration('keyword.control'),
-            decoration('keyword.unsafe'),
-            decoration('function'),
-            decoration('parameter'),
-            decoration('constant'),
-            decoration('type'),
-            decoration('builtin'),
-            decoration('text'),
-            decoration('attribute'),
-            decoration('literal'),
-            decoration('macro'),
-            decoration('variable'),
-            decoration('variable.mut', 'underline'),
-            decoration('field'),
-            decoration('module')
-        ];
+                decoration('comment'),
+                decoration('string'),
+                decoration('keyword'),
+                decoration('keyword.control'),
+                decoration('keyword.unsafe'),
+                decoration('function'),
+                decoration('parameter'),
+                decoration('constant'),
+                decoration('type'),
+                decoration('builtin'),
+                decoration('text'),
+                decoration('attribute'),
+                decoration('literal'),
+                decoration('macro'),
+                decoration('variable'),
+                decoration('variable.mut', 'underline'),
+                decoration('field'),
+                decoration('module')
+            ];
```

Comment:
[Almost](https://github.com/rust-analyzer/rust-analyzer/pull/2061#issuecomment-546325367), with some tweaking I had in mind and if it's cool with you I would offer a mapper in the highlighting section to read from config.

Reply:
@matklad I want to point that I like the tag names and I'd want to keep them around. In the future I am sure there will be a better way of doing what this PR does, so changing them is counter productive as it would be loss of information. 
I think it's just better for now temporary map them when picking out the scopes from the themes & settings loaded.

Reply:
@matklad They do now :)